src/vec/tests.rs

```diff
@@ -227,6 +227,13 @@ fn test_iter() {
     assert_eq!(sv.iter().skip(3).count(), 0);
     assert_eq!(sv.iter().skip(4).count(), 0);
     assert_eq!(sv.iter().skip(usize::MAX).count(), 0);
+
+    {
+        assert_eq!(sv.len(), 3);
+        let mut iter = sv.iter();
+        iter.next_back();
+        assert_eq!(iter.by_ref().count(), 2);
```

Comment:
Before this PR, this returns 3. The stdlib `Vec` returns 2 so we should also return 2.